### PR TITLE
Added required input fields

### LINKED ISSUE #90: Budget Tab Issues
# Do not create single PR for this issue. Divide these issue into part.

- [x] Gray out `Report Fund Spend` on the floating action button. since it is not implemented.
- [x] Whenever in dropdown, if there is no data, then show `No Data available` instead of nothing.
For Refrence:
Screenshot_20200831_111111
- [x] Button Order on every form: First Crete button, then Cancel button and then Reset button.
- [x] When there is no data for nested table, show `No (w/e is the context)  available`. You can look how it is shown in `Deliverables`.
- [x] Note in `Report Budget Spend` is mandatory?  Chekc with the schema. Mandatory fields are marked with *.
- [x] Whenever we are moving to next page on top table pagination, It is showing No Budget Target found instead of loader. Because of this, if the api is taking time, user will no Budget Target availabel instead of loading screen. To solve this, show skeleton while api is in progress. There is a directory in component  called `Skeleton` which contains the skeleton for different component. Create a skeleton which will be used in imapct and deliverable also. Same goes for nested table.
- [x] When we click on `Report Budget Spend`, then on the left side of the popup, the text is `Report Expenditure`. These 2 re completely different. Can you confirm with Himanshu sir it is how it should be?
- [x] After creating Budget Category, it is not showing in Create Budget Target form automatically, Right now, we need to refresh the page to see the changes.
- [ ] After creating Buget Target, why are we refecthing the budget target list instead of updating the cache?

### PR DESCRIPTION
**Pull request by shubhamsingh3211**
Made input fields required and not required

Fixes #90

### CODE REVIEW

File: src/components/Table/Budget/BudgetLineItemTable/BudgetLineItemTable.tsx [hunk 1]
```diff
@@ -73,15 +73,21 @@ const getInitialValues = (
...
-		fy_org: budgetLineItem ? budgetLineItem?.fy_org?.id : "",
-		fy_donor: budgetLineItem ? budgetLineItem?.fy_donor?.id : "",
+		amount: budgetLineItem?.amount ? budgetLineItem?.amount : 0,
+		note: budgetLineItem?.note ? budgetLineItem?.note : "",
+		budget_targets_project: budgetLineItem?.budget_targets_project.id
+			? budgetLineItem?.budget_targets_project.id
+			: "",
+		annual_year: budgetLineItem?.annual_year?.id ? budgetLineItem?.annual_year?.id : "",
+		reporting_date: getTodaysDate(
+			budgetLineItem?.reporting_date ? budgetLineItem?.reporting_date : undefined
```
Comment (anirudh411) [inline src/components/Table/Budget/BudgetLineItemTable/BudgetLineItemTable.tsx:83]: `budgetLineItem?.reporting_date ? budgetLineItem?.reporting_date : undefined` expression like these can be simplified  to ` budgetLineItem?.reporting_date  || undefined`. 
The later one is more more readable.

File: src/components/Table/Budget/BudgetTargetTable/BudgetTargetTableRow.tsx [hunk 2]
```diff
@@ -148,7 +148,9 @@ function BudgetTargetTableRow({
 												subtitle1: "h1",
 											}}
 										>
-											{budgetTarget.description}
+											{budgetTarget?.description
+												? budgetTarget?.description
+												: ""}
```
Comment (anirudh411) [inline src/components/Table/Budget/BudgetTargetTable/BudgetTargetTableRow.tsx:153]: can be simplied

File: src/components/Table/Donor/DonorTable.tsx [hunk 3]
```diff
@@ -47,16 +47,16 @@ const keyNames = ["name", "legal_name", "short_name", "country,name"];
...
-		legal_name: donor ? donor.legal_name : "",
-		name: donor ? donor.name : "",
-		short_name: donor ? donor.short_name : "",
-		id: donor ? donor.id : "",
+		country: donor?.country?.id ? donor?.country?.id : "",
+		legal_name: donor?.legal_name ? donor?.legal_name : "",
+		name: donor?.name  ? donor?.name : "",
+		short_name: donor?.short_name ? donor?.short_name : "",
+		id: donor?.id ? donor?.id : "",
 	};
```
Comment (anirudh411) [inline src/components/Table/Donor/DonorTable.tsx:55]: all can be simplified like this `donor?.country?.id || ''`

File: src/models/index.ts [hunk 4]
```diff
@@ -117,6 +118,7 @@ export interface ISelectField {
 	size: any;
 	hidden?: boolean;
 	displayName?: string;
+	required?: boolean;
 }
```
Comment (anirudh411) [inline src/models/index.ts:123]: Please remove the use of two similar interfaces

File: src/utils/index.tsx [hunk 5]
```diff
@@ -20,3 +20,25 @@ export const compareObjectKeys = (obj1: any, obj2: any): boolean =>
 	Object.keys(obj1).every((key) => obj2.hasOwnProperty(key) && obj2[key] === obj1[key]);
 
 export const isEmptyObject = (obj: object) => Object.keys(obj).length == 0;
+
+interface IObject {
```
Comment (anirudh411) [inline src/utils/index.tsx:24]: redundant type

**Review by anirudh411 — CHANGES_REQUESTED** [review]

File: src/models/index.ts (hunk 4 above)
Comment (shubhamsingh3211) [reply src/models/index.ts:123]: I have already made changes in 17 files and this change will make change in 10 - 15 file, I will make seprate pr for that

**Review by anirudh411 — APPROVED** [review]

**Pull request merged**